iree/base/internal/file_io.c

```diff
@@ -123,21 +132,65 @@ iree_status_t iree_file_write_contents(const char* path,
   return status;
 }
 
+static iree_status_t iree_stdin_read_contents_impl(
+    iree_allocator_t allocator, iree_byte_span_t* out_contents) {
+  // HACK: fix stdin mode to binary on Windows to match Unix behavior.
+  // Ideally we'd do this in one place for all our tools.
+  IREE_SET_BINARY_MODE(stdin);
+
+  iree_host_size_t capacity = 4096;
+  iree_host_size_t size = 0;
+  char* buffer = NULL;
+  IREE_RETURN_IF_ERROR(
+      iree_allocator_malloc(allocator, capacity, (void**)&buffer));
+  for (int c = getchar(); c != EOF; c = getchar()) {
```

Comment:
I was expecting something more sophisticated than getchar(). Fine for now - just likely to be slow...

Reply:
yeah - I really just went for what worked on windows and linux at the same time - improvements very much welcome :)